modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -165,8 +171,10 @@ export default class TileLayer<DataT = any, ExtraPropsT extends {} = {}> extends
   }
 
   get isLoaded(): boolean {
-    return this.state?.tileset?.selectedTiles?.every(
-      tile => tile.isLoaded && tile.layers && tile.layers.every(layer => layer.isLoaded)
+    return Boolean(
+      this.state?.tileset?.selectedTiles?.every(
+        tile => tile.isLoaded && tile.layers && tile.layers.every(layer => layer.isLoaded)
```

Comment:
Q: Is it different from checking if all sublayers are loaded?

Reply:
Yeah it behaves differently, this is checking that the tile data is loaded and then also all layers the tile renders are loaded (it may be many layers per tile). I think sublayers can be in a loaded state while data is still loading so its better to wait for it all.

I don't think this is easy logic to follow.. I attempted to simplify it in https://github.com/visgl/deck.gl/pull/8335#discussion_r1412497221 but it's not so straightforward.